### ISSUE TITLE
confused with poutyne.framework.Model and torch.nn.Module

### ISSUE THREAD

**Issue by menghuu** [GRAAL-Research/poutyne#27]
I know `framework.Model` is:

>    The Model class encapsulates a PyTorch module/network, a PyTorch optimizer,
>    a loss function and metric functions. It allows the user to train a neural
>    network without hand-coding the epoch/step logic.

also, `nn.Module` is not `nn.Model`, but many people implement the `nn.Module` with the name `Model` and create instance with name `model`. It's little confusing when writing callback class (when accessing `self.model`) and the `framework.Model.__init__` has the parameter `model` which is in pytorch's concept.

Can we change the name of `Model` to `Engine` (just like [ignite](https://github.com/pytorch/ignite/blob/38a4f37de759e33bc08441bde99bcb50f3d81f55/ignite/engine/engine.py#L49) or [tnt](https://github.com/pytorch/tnt/blob/3ed904b2cbed16d3bab368bb9ca5adc876d3ce69/torchnet/engine/engine.py#L1)) or other name?

**Comment by freud14** [comment]
Hi,
Sorry for the delay. Maybe we could rename the attribute model to something like "module" or "net/network"? What do you think?
Frédérik

**Comment by menghuu** [comment]
two ways to make the concept clearly

1. change the `framework.Model` to `framework.Engine` or other
2. change the attribute to other name, just like @freud14 said

poutyne's easy-training/evaluation feature should be summary as:
> make a poutyne <1> containing a pytorch <2>, and <1> will train/fit and eval/test pytorch <2>

there are `Model`/`Engine`/`Module`/`Net`/`Network` to select

if poutyne's design aim is a full feature framework like keras+tensorflow, call <1> `Model` may be reasonable. 
in this condition, I think `net/network` should be better than `module` when refer the original pytorch `model`, sorry I am not a native English speaker, `module` just looks like `model` 😓 and normally pytorch `model` consist of many `modules`, `module` is not a good choice to describe the pytorch `model`. but, in other way, `module` object is consistent with class `nn.Module`, it hard to choice `module` or not. calling it `net/network` should not be confusing. (a shortcut is: can `net/network` cover all kinds of `models` in pytorch?)


if the aim is focusing on training or something like this, <1> should not be called with `Model`, many pytorch users have defined their `model/net/network` by pytorch, they only want a easy way to training their pytorch `model/net/network`.
In this condition, changing the poutyne Model to `Engine` or `OtherName` may be great choice, the attribute can be called `model` with no side effect

**Comment by davebulaval** [comment]
@menghuu We will investigate and modify if necessary in the next release. But good call.

**Comment by davebulaval** [comment]
@menghuu I must said you're right it might be confusing to use a model arg/attribute named model when refereing to a network wich implement some PyTorch Module. For myself, I prefer the net/network when refering to a user class using some pytorch module.

**Comment by freud14** [comment]
Thank you for the issue. The name of the `model` attribute is now `network`.

**Comment by freud14** [comment]
See PR #53 and the dev branch for details.
